src/index.ts

```diff
@@ -8,6 +11,17 @@ import { rule as validPackageDef } from "./rules/valid-package-def.js";
 import { rule as validRepositoryDirectory } from "./rules/valid-repository-directory.js";
 import { rule as validVersion } from "./rules/valid-version.js";
 
+declare const TSUP_FORMAT: "cjs" | "esm";
+const req =
+	typeof TSUP_FORMAT === "undefined" || TSUP_FORMAT === "cjs"
+		? require
+		: createRequire(import.meta.url);
```

Comment:
[Refactor] I'm not 100% confident in CJS/ESM shenanigans, but - my instinct is that encoding tsup-specific globals into the source code isn't super long-term tenable. I'd rather rely on checking what exists at build/run-time rather than details of the tool used to build. Is there any reason why the following check of _"either ESM's `import.meta.url` works or CJS's `__filename` will exist"_ wouldn't work?

```suggestion
const require = createRequire(import.meta.url || __filename);
```